src/Shared/Recruit.Vacancies.Client/Infrastructure/Services/EmployerAccount/EmployerAccountProvider.cs

```diff
@@ -85,5 +85,28 @@ public async Task<string> GetEmployerAccountPublicHashedIdAsync(long accountId)
                 throw;
             }
         }
+
+        public async Task<bool> GetEmployerAccountEOIAsync(string accountId)
+        {
+            try
+            {
+                var account = await _accountApiClient.GetAccount(accountId);
+                foreach (var entity in account.LegalEntities)
+                {
+                    var legalEntity = await _accountApiClient.GetLegalEntity(account.HashedAccountId, long.Parse(entity.Id));
+                    foreach (var agreement in legalEntity.Agreements)
```

Comment:
`return legalEntity.Agreements.All(ag => ag.AgreementType == SFA.DAS.Common.Domain.Types.AgreementType.NonLevyExpressionOfInterest)` instead of the inner for each loop